fe/fe-core/src/main/java/org/apache/doris/nereids/types/coercion/DateLikeType.java

```diff
@@ -17,8 +17,30 @@
 
 package org.apache.doris.nereids.types.coercion;
 
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+
 /**
  * date like type.
  */
 public abstract class DateLikeType extends PrimitiveType {
+    private Calendar toCalendar(double d) {
+        //d = (year * 10000 + month * 100 + day) * 1000000L;
+        int date = (int) (d / 1000000);
+        int day = date % 100;
+        int month = (date / 100) % 100;
+        int year = date / 10000;
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.YEAR, year);
+        calendar.set(Calendar.MONDAY, month);
+        calendar.set(Calendar.DAY_OF_MONTH, day);
```

Comment:
What about hours, minutes and seconds?

Reply:
currently, only support date/datev2.
this function can be extended to support time if we need

Reply:
I will add a TODO to note it

Reply:
No you'd better do it right now,` TODO` often means it will never get done